Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.ColorDialog.cs

```diff
@@ -0,0 +1,156 @@
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics.Contracts;
+using System.Drawing;
+using System.Runtime.InteropServices;
+using System.Security.Permissions;
+
+namespace System.Windows.Forms
+{
+    // <summary>
+    // Represents a common dialog box that displays available colors along with controls that enable the user to define custom colors.
+    // </summary>
+    // <filterpriority>1</filterpriority>
+    public class ColorDialog : CommonDialog
+    {
+        // <summary>
+        // Gets or sets a value indicating whether the user can use the dialog box to define custom colors.
+        // </summary>
+        //
+        // <returns>
+        // true if the user can define custom colors; otherwise, false. The default is true.
+        // </returns>
+        // [DefaultValue(true)]
+        // public virtual bool AllowFullOpen { get; set; }
+        
+        // <summary>
+        // Gets or sets a value indicating whether the dialog box displays all available colors in the set of basic colors.
+        // </summary>
+        // <returns>
+        // true if the dialog box displays all available colors in the set of basic colors; otherwise, false. The default value is false.
+        // </returns>
+        // public virtual bool AnyColor {get; set;}
+       
+        // <summary>
+        // Gets or sets the color selected by the user.
+        // </summary>
+        //
+        // <returns>
+        // The color selected by the user. If a color is not selected, the default value is black.
+        // </returns>
+        // public Color Color {get; set;}
+       
+        /// <summary>
+        /// Gets or sets the set of custom colors shown in the dialog box.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// A set of custom colors shown by the dialog box. The default value is null.
+        /// </returns>
+        public int[] CustomColors
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<int[]>() != null);
+                return default(int[]);
+            }
+            set
```

Comment:
Can the empty contactless setter be omitted, or is this required in order to match the signature of the framework?